kmk/kmk_keyboard.py

```diff
@@ -1,9 +1,15 @@
+try:
+    from typing import Optional
+except ImportError:
+    pass
```

Comment:
I'm not aware of a situation where this import can fail _and_ inline type annotations will be implemented. I'd imagine any Python implementation going through the effort to implement the lexer/parser bits would also implement the stdlib bits?

"version-resilient" type annotations in Python take the comment form, but we decided in https://github.com/KMKfw/kmk_firmware/pull/501 to just bump minimum CPy version to 7.x, which implements the non-comment annotations anyway.

Reply:
I've got CPy v7.3.2 on my test board, and it doesn't have the `typing` builtin.

Reply:
and the ref to Optional is.... I suppose ignored by the parser later on, then?
 
if it runs, :shipit: , interesting tradeoffs but it's a microcontroller, I get it

Reply:
Just for info: I checked a couple of repos, and that's the idiom official adafruit libs use... Guess it's the best we can do for now.